packages/web/hooks/limit-orders/use-orderbook-pool.ts

```diff
@@ -1,15 +1,43 @@
+import { Dec } from "@keplr-wallet/unit";
+import { useMemo, useState } from "react";
+
+import { api } from "~/utils/trpc";
+
 export const useOrderbookPool = ({
   baseDenom,
   quoteDenom,
 }: {
   baseDenom: string;
   quoteDenom: string;
 }) => {
+  const [contractAddress] = useState<string>(
+    "osmo1svmdh0ega4jg44xc3gg36tkjpzrzlrgajv6v6c2wf0ul8m3gjajs0dps9w"
+  );
+  const { makerFee, isLoading: isMakerFeeLoading } = useMakerFee({
+    orderbookAddress: contractAddress,
+  });
   return {
     poolId: "1",
     baseDenom,
     quoteDenom,
-    orderbookContractAddress:
-      "osmo1svmdh0ega4jg44xc3gg36tkjpzrzlrgajv6v6c2wf0ul8m3gjajs0dps9w",
+    contractAddress,
+    makerFee,
+    isMakerFeeLoading,
+  };
+};
+
+const useMakerFee = ({ orderbookAddress }: { orderbookAddress: string }) => {
+  const { data: makerFeeData, isLoading } =
+    api.local.orderbook.getMakerFee.useQuery({
+      contractOsmoAddress: orderbookAddress,
+    });
+
+  const makerFee = useMemo(() => {
+    if (isLoading) return new Dec(0);
+    return new Dec(makerFeeData?.makerFee ?? 0);
+  }, [isLoading, makerFeeData]);
+  return {
+    makerFee,
+    isLoading,
```

Comment:
Why not do this dec conversion in the tRPC router? It would likely be useful to other users of that function in the future. tRPC is set up with superjson so it can send Dec, PricePretty, CoinPretty, Int, etc. over the wire.

Reply:
https://github.com/osmosis-labs/osmosis-frontend/pull/3321/commits/371934ab9171d5c79d1bc900cbc9e13056217acf